src/premiumvideodevice/PremiumVideoTransformDevice.ts

```diff
@@ -0,0 +1,91 @@
+import Device from '../devicecontroller/Device';
+import VideoTransformDevice from '../devicecontroller/VideoTransformDevice';
+import Logger from '../logger/Logger';
+import PremiumVideoStreamHandler from './PremiumVideoStreamHandler';
+import PremiumVideoEffectDriver from './PremiumVideoEffectDriver';
+
+/**
+ * [[PremiumVideoTransformDevice]] will allow us to transform a regular device
+ * into a format that will allow us to intercept it's MediaStream via a
+ * [[PremiumVideoStreamHandler]] and apply frame-level effects via a
+ * [[PremiumVideoEffectDriver]]
+ */
+export default class PremiumVideoTransformDevice implements VideoTransformDevice {
+  private inputMediaStream: MediaStream;
+  private premiumVideoStreamHandler: PremiumVideoStreamHandler;
+
+  constructor(
+    private logger: Logger,
+    private device: Device,
+    private videoEffectDriver: PremiumVideoEffectDriver
+  ) {
+    // initialize the stream handler
+    this.premiumVideoStreamHandler = new PremiumVideoStreamHandler(logger, videoEffectDriver);
+  }
+
+  /**
+   * remove access to a media stream and stop all running video tracks
+   */
+  async stop(): Promise<void> {
+    if (this.inputMediaStream) {
```

Comment:
Sorry I missed this but this can be `this.inputMediaStream?.getVideoTracks().map(track => track.stop());`